components/Modal/CreateImageUpload.js

```diff
@@ -442,88 +496,297 @@ class CreateImageUploadModal extends React.Component {
+                      values={{
+                        iam: <strong>Identity and Access Management (IAM)</strong>
+                      }}
+                    />
+                  }
+                  aria-label="secret access key help"
+                >
+                  <Button variant="plain" aria-label="secret access key help">
+                    <OutlinedQuestionCircleIcon id="popover-icon" />
+                  </Button>
+                </Popover>
+              </div>
+              <TextInput
+                className="pf-c-form-control"
+                value={this.state.uploadSettings["secretAccessKey"]}
+                type="password"
+                id="secret-access-key-input"
+                name="secretAccessKey"
+                onChange={this.setUploadSettings}
+              />
+            </div>
+          </Form>
         </React.Fragment>
       )
-    });
+    };
 
-    const uploadSettings = provider => ({
-      name: "File upload",
+    const awsUploadSettings = {
+      name: "Destination",
       component: (
         <React.Fragment>
           <Text className="help-block cc-c-form__required-text">
             <FormattedMessage defaultMessage="All fields are required." />
           </Text>
-          {providerSettings[provider] !== undefined && (
-            <Form isHorizontal>
-              <FormGroup label={formatMessage({ id: "image-name", defaultMessage: "Image name" })} fieldId="image-name">
-                <TextInput
-                  value={imageName}
-                  type="text"
-                  id="image-name"
-                  aria-describedby="image-name"
-                  onChange={() => this.setState({ imageName: event.target.value })}
-                  isRequired
-                />
-              </FormGroup>
-              {Object.keys(providerSettings[provider].settings).map(key => (
-                <FormGroup
-                  label={formatMessage({
-                    id: `${provider}-settings`,
-                    defaultMessage: providerSettings[provider].settings[key].displayText
-                  })}
-                  fieldId={key}
-                  key={key}
+          <Form isHorizontal className="cc-m-wide-label">
+            <div className="pf-c-form__group">
+              <div className="pf-c-form__label pf-m-no-padding-top pf-l-flex pf-u-display-flex pf-m-justify-content-flex-start pf-m-nowrap">
+                <label htmlFor="image-name-input" className="pf-l-flex__item">
+                  <span className="pf-c-form__label-text">
+                    <FormattedMessage defaultMessage="Image name" />
+                  </span>
+                  <span className="pf-c-form__label-required" aria-hidden="true">
+                    &#42;
+                  </span>
+                </label>
+                <Popover
+                  id="popover-help"
```

Comment:
Could you please give `id` a unique name, like `image-name popover`? That'll be very helpful to find this element when write test code. Thanks!